packages/core/src/extensions/FilePanel/FilePanelPlugin.ts

```diff
@@ -1,206 +1,79 @@
+import { Plugin } from "@tiptap/pm/state";
+
+export const FilePanelPlugin = createExtension((editor) => {
+  const store = createStore({
+    blockId: undefined as string | undefined,
+  });
+
+  function closeMenu() {
+    store.setState({
+      blockId: undefined,
+    });
   }
 
-  private view: FilePanelView<I, S> | undefined;
-
-  constructor(editor: BlockNoteEditor<Record<string, any>, I, S>) {
-    super();
-    this.addProsemirrorPlugin(
-      new Plugin<{
-        block: BlockFromConfig<any, I, S> | undefined;
-      }>({
-        key: filePanelPluginKey,
-        view: (editorView) => {
-          this.view = new FilePanelView<I, S>(
-            editor,
-            filePanelPluginKey as any,
-            editorView,
-            (state) => {
-              this.emit("update", state);
-            },
-          );
-          return this.view;
-        },
+  // reset the menu when the document changes
+  editor.onChange(
+    closeMenu,
+    // don't trigger the callback if the changes are caused by a remote user
+    false,
+  );
+
+  // reset the menu when the selection changes
+  editor.onSelectionChange(closeMenu);
+
+  return {
+    key: "filePanel",
+    store,
+    closeMenu,
+    plugins: [
+      // TODO annoying to have to do this here
+      new Plugin({
         props: {
           handleKeyDown: (_view, event: KeyboardEvent) => {
-            if (event.key === "Escape" && this.shown) {
-              this.view?.closeMenu();
+            if (event.key === "Escape" && store.state.blockId) {
+              closeMenu();
               return true;
             }
             return false;
           },
         },
-        state: {
-          init: () => {
-            return {
-              block: undefined,
-            };
-          },
-          apply: (transaction, prev) => {
-            const state: FilePanelState<I, S> | undefined =
-              transaction.getMeta(filePanelPluginKey);
-
-            if (state) {
-              return state;
-            }
-
-            if (
-              !transaction.getMeta(ySyncPluginKey) &&
-              (transaction.selectionSet || transaction.docChanged)
-            ) {
-              return { block: undefined };
-            }
-            return prev;
-          },
-        },
       }),
-    );
-  }
-
-  public get shown() {
-    return this.view?.state?.show || false;
-  }
-
-  public onUpdate(callback: (state: FilePanelState<I, S>) => void) {
-    return this.on("update", callback);
-  }
-
-  public closeMenu = () => this.view?.closeMenu();
-}
+    ],
+    showMenu(blockId: string) {
+      store.setState({
+        blockId,
+      });
+    },
+    init({ dom, root, abortController }) {
+      dom.addEventListener("mousedown", closeMenu, {
+        signal: abortController.signal,
+      });
+      dom.addEventListener("dragstart", closeMenu, {
+        signal: abortController.signal,
+      });
+
+      root.addEventListener(
+        "scroll",
```

Comment:
my guess is that this was previously here to update the position on scroll, but that floatingui can handle this now indeed